File: C++/pc.cpp
#include <string>
#include "processor.cpp"
#include "disk.cpp"
#include "ram.cpp"

using namespace std;

class PC{
    private:
        //atribut
        Processor processor;
        disk Disk;
        ram RAM;
        int totalPrice;

    public:
        //konstruktor
        PC() { }

        PC(Processor processor, disk Disk, ram RAM, int totalPrice){
            this->processor = processor;
            this->Disk = Disk;
            this->RAM = RAM;
            this->totalPrice = processor.getPrice() + Disk.getPrice() + RAM.getPrice();
        }

        // get dan set pc
        void setprocessor(Processor processor){
            this->processor = processor;
        }

        Processor getprocessor(){
            return this->processor;
        }

        void setDisk(disk Disk){
            this->Disk = Disk;
        }

        disk getDisk(){
            return this->Disk;
        }

        void setRam(ram RAM){
            this->RAM = RAM;
        }

        ram getRam(){
            return this->RAM;
        }

        void setTotalPrice(int Procprice, int diskPrice, int ramPrice){
            this->totalPrice = Procprice + diskPrice + ramPrice;
        }

        int getTotalPrice(){
            return this->totalPrice;
        }

        ~PC(){ }

};

File: PHP/pc.php
<?php

    class pc
    {
        private $processor;
        private $disk;
        private $ram;
        private $totalPrice = 0;
        
        #konstruktor
        public function __construct($processor, $disk, $ram, $totalPrice = 0){
            $this->processor = $processor;
            $this->disk = $disk;
            $this->ram = $ram;
            $this->totalPrice = $totalPrice;
        }

        #prosedur setter
        public function setProcessor($processor){
          $this->processor = $processor;  
        }

        public function setDisk($disk){
            $this->disk = $disk;
        }

        public function setRam($ram){
            $this->ram = $ram;
        }

        public function totalPrice($totalPrice){
            $this->totalPrice = $totalPrice;
        }

        #prosedur getter
        public function getProcessor(){
            return $this->processor;
        }

        public function getDisk(){
            return $this->disk;
        }

        public function getRam(){
            return $this->ram;
        }

        public function getTotalPrice(){
            return $this->totalPrice;
        }

        public function __destruct(){ }


    }
?>